File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    p1 = p2 = len(sequence)-1
    current_sum = sequence[-1]
    
    while current_sum != k :
        if current_sum > k :
            if p1 == p2:
                p1-=1
                current_sum+=sequence[p1]
            current_sum-=sequence[p2]
            p2-=1
        elif current_sum < k :
            p1-=1
            current_sum+=sequence[p1]
            
    while sequence[p2] == sequence[p1-1] and p1 > 0 :
        p1, p2 = p1-1, p2-1
        
    return [p1, p2]

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let p1 = sequence.length-1;
    let p2 = sequence.length-1;
    let sum = sequence[sequence.length-1];
    while(sum!==k){
        if(sum>k){
            if(p1===p2){
                p1--;
                sum+=sequence[p1];
            }
            sum-=sequence[p2];
            p2--;
        } else if(sum<k){
            p1--;
            sum+=sequence[p1];
        }
    }
    while(sequence[p2] === sequence[p1-1]){
        p1--;
        p2--;
    }
    return [p1, p2];
}